src/Dapr.Workflow/WorkflowContext.cs

```diff
@@ -44,47 +44,299 @@ internal WorkflowContext(TaskOrchestrationContext innerContext)
+        /// that started it.
+        /// </para><para>
+        /// Child workflows have many benefits:
+        /// <list type="bullet">
+        ///  <item>You can split large workflows into a series of smaller child workflows, making your code more maintainable.</item>
+        ///  <item>You can distribute workflow logic across multiple compute nodes concurrently, which is useful if
+        ///  your workflow logic otherwise needs to coordinate a lot of tasks.</item>
+        ///  <item>You can reduce memory usage and CPU overhead by keeping the history of parent workflow smaller.</item>
+        /// </list>
+        /// </para><para>
+        /// The return value of a child workflow is its output. If a child workflow fails with an exception, then that
+        /// exception will be surfaced to the parent workflow, just like it is when an activity task fails with an
+        /// exception. Child workflows also support automatic retry policies.
+        /// </para><para>
+        /// Because child workflows are independent of their parents, terminating a parent workflow does not affect
+        /// any child workflows. You must terminate each child workflow independently using its instance ID, which is
+        /// specified by supplying <see cref="SubOrchestrationOptions" /> in place of <see cref="TaskOptions" />.
+        /// </para>
+        /// </remarks>
+        /// <param name="workflowName">The name of the workflow to call.</param>
+        /// <param name="input">The serializable input to pass to the child workflow.</param>
+        /// <param name="options">
+        /// Additional options that control the execution and processing of the child workflow. Callers can choose to
+        /// supply the derived type <see cref="SubOrchestrationOptions" />.
+        /// </param>
+        /// <returns>A task that completes when the child workflow completes or fails.</returns>
+        /// <exception cref="ArgumentException">The specified workflow does not exist.</exception>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if the calling thread is not the workflow dispatch thread.
+        /// </exception>
+        /// <exception cref="TaskFailedException">
+        /// The child workflow failed with an unhandled exception. The details of the failure can be found in the
+        /// <see cref="TaskFailedException.FailureDetails"/> property.
+        /// </exception>
+        public Task CallChildWorkflowAsync(string workflowName, object? input = null, TaskOptions? options = null)
+        {
+            return this.innerContext.CallSubOrchestratorAsync(workflowName, input, options);
+        }
+
+        /// <summary>
+        /// Restarts the workflow with a new input and clears its history.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This method is primarily designed for eternal workflows, which are workflows that
```

Comment:
Are eternal workflows common? The idea of that scares me 😅 I'm fine keeping this, just curious.

Reply:
Yes, it's actually very common. 😅 It's a way for you to create some kind of long running agent that doesn't have an explicit lifetime, like a monitor that maybe runs in a loop and only takes action when some event occurs.